From 67d113a76162e85c3ee75d76219a66359656087d Mon Sep 17 00:00:00 2001
From: Paolo Bonzini <pbonzini@redhat.com>
Date: Thu, 18 Jun 2015 11:42:44 +0200
Subject: Fix rpmdiff failure

Patchwork-id: 66323
O-Subject: [RHEL7.2 PATCH libunwind] Fix rpmdiff failure
Bugzilla: 1229359
RH-Acked-by: Miroslav Rezanina <mrezanin@redhat.com>
RH-Acked-by: Jeff Nelson <jenelson@redhat.com>

This fixes an rpmdiff failure for libunwind.  It is possible to build
a cross libunwind to e.g. work on ARM memory from x86.  Because of this,
the libunwind.h file differs across architectures and rpmdiff complains
because it is installed in /usr/include.

RHEL never does cross libunwinds, so just drop the faulty include
statement.
---
 include/libunwind.h.in | 13 ++++++-------
 1 file changed, 6 insertions(+), 7 deletions(-)

diff --git a/include/libunwind.h.in b/include/libunwind.h.in
index 7a56168..aad564c 100644
--- a/include/libunwind.h.in
+++ b/include/libunwind.h.in
@@ -1,7 +1,11 @@
 /* Provide a real file - not a symlink - as it would cause multiarch conflicts
    when multiple different arch releases are installed simultaneously.  */
 
-#ifndef UNW_REMOTE_ONLY
+/* Fedora/RHEL does not cross-compile libunwind, so always use direct
+ * inclusion.  It handles multilib better.
+ */
+#ifndef LIBUNWIND_INCLUDE_H
+#define LIBUNWIND_INCLUDE_H 1
 
 #if defined __aarch64__
 #include "libunwind-aarch64.h"
@@ -28,9 +32,4 @@
 #else
 # error "Unsupported arch"
 #endif
-
-#else /* UNW_REMOTE_ONLY */
-
-# include "libunwind-@arch@.h"
-
-#endif /* UNW_REMOTE_ONLY */
+#endif /* LIBUNWIND_INCLUDE_H */
-- 
1.8.3.1

